File: trunk/RL-Glue/Java/src/rlglue/environment/Environment.java
package rlglue.environment;

import rlglue.types.Action;
import rlglue.types.Observation;
import rlglue.types.Random_seed_key;
import rlglue.types.Reward_observation;
import rlglue.types.State_key;

public interface Environment
{
    String env_init();
    Observation env_start();
    Reward_observation env_step(Action action);
    void env_cleanup();
    void env_set_state(State_key key);
    void env_set_random_seed(Random_seed_key key);
    State_key env_get_state();
    Random_seed_key env_get_random_seed();
    String env_message(final String message);
}

File: trunk/RL-Glue/Python/rlglue/environment/Environment.py

from rlglue.types import Action
from rlglue.types import Observation
from rlglue.types import Random_seed_key
from rlglue.types import Reward_observation
from rlglue.types import State_key

class Environment:
	# () -> string
	def env_init():
		pass
	
	# () -> Observation
 	def env_start():
		pass
	
	# (Action) -> Reward_observation
	def env_step(action):
		pass
	
	# () -> void
	def env_cleanup():
		pass
	
	# (State_key) -> void
	def env_set_state(key):
		pass
	
	# (Random_seed_key) -> void
	def env_set_random_seed(key):
		pass
	
	# () -> State_key
	def env_get_state():
		pass
	
	# () -> Random_seed_key
	def env_get_random_seed():
		pass
	
	# (string) -> string
	def env_message(message):
		pass
